src/modint.rs

```diff
@@ -793,20 +793,23 @@ trait InternalImplementations: ModIntBase {
     #[inline]
     fn add_impl(lhs: Self, rhs: Self) -> Self {
         let modulus = Self::modulus();
-        let mut val = lhs.val() + rhs.val();
-        if val >= modulus {
-            val -= modulus;
-        }
+        let val = match lhs.val().overflowing_add(rhs.val()) {
```

Comment:
It may be necessary to measure whether it is more efficient to perform a cast operation from u32 to u64 for the input value of the addition and then attempt to subtract in modulo.

https://rust.godbolt.org/z/3z8WsebMo

before code shim:
```rust
pub fn add_impl_shim_before(lhs: u32, rhs: u32, modulus: u32) -> u32 {
    let mut val = lhs + rhs;
    if val >= modulus {
        val -= modulus;
    }
    val
}
```

current PR code shim:
```rust
pub fn add_impl_shim_after(lhs: u32, rhs: u32, modulus: u32) -> u32 {
    match lhs.overflowing_add(rhs) {
        (v, true) => v.wrapping_sub(modulus),
        (v, false) => match v.overflowing_sub(modulus) {
            (_, true) => v,
            (w, false) => w,
        },
    }
}
```

use u32 to u64 cast operation:
```rust
pub fn add_impl_shim_after2(lhs: u32, rhs: u32, modulus: u32) -> u32 {
    let v = (lhs as u64) + (rhs as u64);
    match v.overflowing_sub(modulus as u64) {
        (_, true) => v as u32,
        (w, false) => w as u32,
    }
}
```

Reply:
A benchmark comparison of the proposed implementations of ModInt addition showed no significant difference between the two methods. Since the latter implementation seems easier to understand, we would like to change to the latter.

1. (`add_impl_shim_after`) detect additive overflow (u32) and then compare with modulus (u32)
2. (`add_impl_shim_after2`) cast to u64, then add, compare with modulus(u64)